### PR TITLE
Updated event format

### PR DESCRIPTION
**Pull request by cdahlqvist**
Updated the generated event format to more closely resemble what is generated by the Filebeat Nginx module. Existing dashboards have been updated/modified and a new option to simulate Discover queries added.

### CODE REVIEW

**Comment by cdahlqvist** [comment]
This PR addresses the following issues:

- [rally-eventdata-track #11](https://github.com/elastic/rally-eventdata-track/issues/11)
- [rally-eventdata-track #12](https://github.com/elastic/rally-eventdata-track/issues/12)
- [rally-eventdata-track #14](https://github.com/elastic/rally-eventdata-track/issues/14)

File: eventdata/elasticlogs-index.json [hunk 1]
```diff
@@ -1,45 +1,173 @@
...
+              "ignore_above": 1024
+            }
           }
         }
-      },
-      "_all": { "enabled": false }
+      }
     }
   }
 }
```
Comment (danielmitterdorfer) [inline eventdata/elasticlogs-index.json]: nit: missing new line

File: eventdata/mappings.json [hunk 2]
```diff
@@ -1,43 +1,199 @@
...
         }
       }
-    },
-    "_all": { "enabled": false }
+    }
   }
-}
\ No newline at end of file
+}
+
```
Comment (danielmitterdorfer) [inline eventdata/mappings.json]: nit: trailing new lines

File: eventdata/operations/generate-historic-data.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
   "param-source": "elasticlogs_bulk",
   "index": "elasticlogs-h-<yyyy>-<mm>-<dd>",
   "daily_index": true,
-  "starting_point": "2017-06-01:00:00:00",
-  "acceleration_factor": 6,
+  "starting_point": "2018-05-01:00:00:00",
+  "acceleration_factor": 12,
   "bulk-size": 1000
 }
```
Comment (danielmitterdorfer) [inline eventdata/operations/generate-historic-data.json]: nit: missing new line

File: eventdata/operations/indexing.json [hunk 4]
```diff
@@ -190,4 +209,9 @@
   "name": "deleteindex_elasticlogs_q-*",
   "operation-type": "delete-index",
   "index": "elasticlogs_q-*"
+},
+{
+  "name": "deleteindex_elasticlogs",
+  "operation-type": "delete-index",
+  "index": "elasticlogs"
 }
```
Comment (danielmitterdorfer) [inline eventdata/operations/indexing.json]: nit: missing new line

File: eventdata/operations/indexing.json [hunk 5]
```diff
@@ -115,6 +115,25 @@
...
+{
+  "name": "create_elasticlogs_index",
+  "operation-type": "createindex",
+  "index_name": "elasticlogs",
+  "index_template_body": {
+    "template": "elasticlogs",
+    "settings": {
+      "index.refresh_interval": "5s",
+      "index.codec": "best_compression",
+      "index.number_of_replicas": {{ replica_count | default(0) }},
```
Comment (danielmitterdorfer) [inline eventdata/operations/indexing.json]: Inconsistent naming? In the README for [rally-tracks](https://github.com/elastic/rally-tracks/tree/master/eventdata#parameters) it says `number_of_replicas` (I know that technically this is not the same track...)

File: eventdata/operations/indexing.json [hunk 6]
```diff
@@ -115,6 +115,25 @@
   },
   "index_template_name": "elasticlogs_i_write"
 },
+{
+  "name": "create_elasticlogs_index",
+  "operation-type": "createindex",
```
Comment (danielmitterdorfer) [inline eventdata/operations/indexing.json]: I'm not a 100% sure but I wonder whether you can use the core Rally operation [`create-index`](http://esrally.readthedocs.io/en/stable/track.html#create-index) instead.

File: eventdata/parameter_sources/elasticlogs_bulk_source.py [hunk 7]
```diff
@@ -86,7 +80,7 @@ def params(self):
         bulk_array = []
         for x in range(0, self._bulk_size):
             evt, idx, typ = self._randomevent.generate_event()
-            bulk_array.append('{"index": {"_index": "%s", "_type": "%s"}}"' % (idx, typ))
+            bulk_array.append('{"index": {"_index": "%s", "_type": "doc"}}"' % (idx))
```
Comment (danielmitterdorfer) [inline eventdata/parameter_sources/elasticlogs_bulk_source.py]: nit: if there is only one parameter in the format string, you can remove the parenthesis.

File: eventdata/mappings.json (hunk 2 above)
Comment (cdahlqvist) [reply eventdata/mappings.json]: Always forget that....

File: eventdata/operations/indexing.json (hunk 5 above)
Comment (cdahlqvist) [reply eventdata/operations/indexing.json]: Will align this

File: eventdata/operations/indexing.json (hunk 6 above)
Comment (cdahlqvist) [reply eventdata/operations/indexing.json]: That may be possible, but I will leave it as is for now.

Comment (danielmitterdorfer) [reply eventdata/operations/indexing.json]: Makes sense.

**Comment by cdahlqvist** [comment]
@danielmitterdorfer Have addressed most of your review comments and fixed a bug with the device reference data.

**Pull request merged**